src/screens/Survey/index.test.tsx

```diff
@@ -16,22 +23,67 @@ describe('SurveyScreen', () => {
     );
   };
 
-  it('renders Survey screen and its components', () => {
-    render(<TestComponent />);
-
-    const backgroundImage = screen.getByTestId(mainViewTestIds.base);
-    const backButton = screen.getByTestId(surveyScreenTestIds.backButton);
-    const coverImage = screen.getByTestId(surveyScreenTestIds.coverImage);
-    const title = screen.getByTestId(surveyScreenTestIds.title);
-    const description = screen.getByTestId(surveyScreenTestIds.description);
-    const startSurveyButton = screen.getByTestId(surveyScreenTestIds.startSurveyButton);
-
-    expect(backgroundImage).toBeVisible();
-    expect(backButton).toBeVisible();
-    expect(coverImage).toBeVisible();
-    expect(title).toBeVisible();
-    expect(description).toBeVisible();
-    expect(startSurveyButton).toBeVisible();
-    expect(startSurveyButton).toHaveTextContent('Start Survey');
+  const mockState: { survey: SurveyState } = {
+    survey: {
+      survey: {
+        id: faker.string.uuid(),
+        resourceType: 'survey',
+        title: faker.string.sample(),
+        description: faker.string.sample(),
+        coverImageUrl: faker.image.url(),
+      },
+      isLoading: true,
+      isError: false,
+    },
+  };
+
+  beforeEach(() => {
+    (useAppSelector as jest.Mock).mockImplementation((callback) => callback(mockState));
+    (useAppDispatch as jest.Mock).mockImplementation(() => mockDispatch);
+    jest.spyOn(React, 'useEffect').mockImplementation(() => jest.fn());
```

Comment:
why do we need to mock the `useEffect`?

Reply:
@nvminhtue ah yeah, we can remove it, thanks 🙏